tests/31sync/09archived.pl

```diff
@@ -56,16 +55,67 @@
       })->then( sub {
          my ( $body ) = @_;
 
+         log_if_fail "sync result", $body;
+
          my $room = $body->{rooms}{leave}{$room_id};
          assert_json_keys( $room, qw( timeline state ));
 
          @{ $room->{state}{events} } == 0
             or die "Expected no state events";
 
+         # we should see our own leave event
+         assert_eq(scalar @{ $room->{timeline}{events} }, 1,
+                   "timeline events");
+         my $ev = $room->{timeline}{events}[0];
+         assert_eq($ev->{type}, "m.room.member", "event type");
+         assert_eq($ev->{content}{membership}, "leave", "membership");
+
```

Comment:
argh. This is the problem with having completely different coding styles to every other project. Now fixed.